ttnn/cpp/ttnn/operations/pool/generic/device/pool_multi_core_program_factory.cpp

```diff
@@ -392,7 +392,20 @@ Pool2D::MultiCore::cached_program_t pool2d_multi_core_sharded_with_halo_v2_impl_
 
     auto compute_kernel = CreateKernel(program, compute_kernel_fname, core_range, compute_config);
 
-    // Capture reader_indices_storage to cache this with the program
+    auto temparory_size = program.get_cb_memory_size();
```

Comment:
pls don't use auto everywhere, especially if type is not clear from right hand side